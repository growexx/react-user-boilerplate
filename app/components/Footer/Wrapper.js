import styled from 'styled-components';
const Wrapper = styled.footer`
  display: flex;
  flex-wrap: wrap;
  background: @white;
  justify-content: space-between;
  align-items: center;
  font-size: @font-size-base;
  color: @black;
  padding: @padding-xs 0;
  font-weight: @font-weight-bold;
  > section:first-child {
    margin-left: @margin-xs;
  }
  > section:nth-child(2) {
    margin-right: @margin-xs;
    option {
      color: @black;
    }
    @media only screen and (max-width: 767px) {
      margin-top: @font-size-sm;
    }
  }
`;

export default Wrapper;
