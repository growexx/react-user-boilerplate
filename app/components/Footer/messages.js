/*
 * Footer Messages
 *
 * This contains all the text for the Footer component.
 */
import { defineMessages } from 'react-intl';

export const scope = 'boilerplate.components.Footer';

export default defineMessages({
  copyRightMessage: {
    id: `${scope}.copyRightMessage.message`,
    defaultMessage: `&copy; ${new Date().getFullYear()} React Boilerplate`,
  },
  copyRightSubMessage: {
    id: `${scope}.copyRightSubMessage.message`,
    defaultMessage: `Get started with your next react application in seconds.`,
  },
});
