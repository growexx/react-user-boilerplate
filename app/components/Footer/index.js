import React from 'react';
import { FormattedHTMLMessage } from 'react-intl';
import LocaleToggle from 'containers/LocaleToggle';
import Wrapper from './Wrapper';
import messages from './messages';

function Footer() {
  return (
    <Wrapper>
      <section>
        <FormattedHTMLMessage {...messages.copyRightMessage} />
        <section>
          <FormattedHTMLMessage {...messages.copyRightSubMessage} />
        </section>
      </section>
      <section>
        <LocaleToggle />
      </section>
    </Wrapper>
  );
}

export default Footer;
