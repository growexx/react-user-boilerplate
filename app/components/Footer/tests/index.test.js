import React from 'react';
import renderer from 'react-test-renderer';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';
import { PersistGate } from 'redux-persist/integration/react';
// import history from 'utils/history';

import Footer from '../index';
import configureStore from '../../../configureStore';
let prevStore;
let prevPersistor;

describe('<Footer />', () => {
  beforeAll(() => {
    // store = configureStore({}, browserHistory);
    const { store, persistor } = configureStore({}, browserHistory);
    prevStore = store;
    prevPersistor = persistor;
  });

  it('should render and match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <Provider store={prevStore}>
          <PersistGate persistor={prevPersistor}>
            <IntlProvider locale="en">
              <Footer />
            </IntlProvider>
          </PersistGate>
        </Provider>,
      )
      .toJSON();

    expect(renderedComponent).toMatchSnapshot();
  });
});
