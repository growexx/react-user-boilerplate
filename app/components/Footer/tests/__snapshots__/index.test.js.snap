// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render and match the snapshot 1`] = `
.c2 {
  line-height: 1em;
  background-color: transparent;
  border-style: none;
}

.c1 {
  padding: 2px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  background: #fff;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 14px;
  color: #000;
  padding: 8px 0;
  font-weight: 700;
}

.c0 > section:first-child {
  margin-left: 8px;
}

.c0 > section:nth-child(2) {
  margin-right: 8px;
}

.c0 > section:nth-child(2) option {
  color: #000;
}

@media only screen and (max-width:767px) {
  .c0 > section:nth-child(2) {
    margin-top: 12px;
  }
}

<footer
  className="c0"
>
  <section>
    <span
      dangerouslySetInnerHTML={
        Object {
          "__html": "&copy; 2021 React Boilerplate",
        }
      }
    />
    <section>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "Get started with your next react application in seconds.",
          }
        }
      />
    </section>
  </section>
  <section>
    <div
      className="c1"
    >
      <select
        className="c2"
        onChange={[Function]}
        value="en"
      >
        <option
          value="en"
        >
          en
        </option>
        <option
          value="de"
        >
          de
        </option>
      </select>
    </div>
  </section>
</footer>
`;
